src/platform/silabs/multi-ota/OTAHooks.cpp

```diff
@@ -22,33 +22,44 @@
 #include <app/clusters/ota-requestor/OTARequestorInterface.h>
 
 #include <platform/silabs/multi-ota/OTAFactoryDataProcessor.h>
+#ifndef SLI_SI91X_MCU_INTERFACE
 #include <platform/silabs/multi-ota/OTAFirmwareProcessor.h>
+#endif
+
+#if SL_WIFI
+#include <platform/silabs/multi-ota/SiWx917/OTAWiFiFirmwareProcessor.h>
+#endif
 
 #if OTA_TEST_CUSTOM_TLVS
 #include <platform/silabs/multi-ota/OTACustomProcessor.h>
 #endif
 
 CHIP_ERROR chip::OTAMultiImageProcessorImpl::ProcessDescriptor(void * descriptor)
 {
+#ifndef SLI_SI91X_MCU_INTERFACE
     [[maybe_unused]] auto desc = static_cast<chip::OTAFirmwareProcessor::Descriptor *>(descriptor);
     ChipLogDetail(SoftwareUpdate, "Descriptor: %ld, %s, %s", desc->version, desc->versionString, desc->buildDate);
+#endif
+
+#if SL_WIFI
+    auto descWiFi = static_cast<chip::OTAWiFiFirmwareProcessor::Descriptor *>(descriptor);
+    ChipLogDetail(SoftwareUpdate, "Descriptor: %ld, %s, %s", descWiFi->version, descWiFi->versionString, descWiFi->buildDate);
+#endif
 
     return CHIP_NO_ERROR;
 }
 
 CHIP_ERROR chip::OTAMultiImageProcessorImpl::OtaHookInit()
 {
-    static chip::OTAFirmwareProcessor sApplicationProcessor;
-    static chip::OTAFactoryDataProcessor sFactoryDataProcessor;
-
-    sApplicationProcessor.RegisterDescriptorCallback(ProcessDescriptor);
-    sFactoryDataProcessor.RegisterDescriptorCallback(ProcessDescriptor);
 
     auto & imageProcessor = chip::OTAMultiImageProcessorImpl::GetDefaultInstance();
+
+#ifndef SLI_SI91X_MCU_INTERFACE
+    static chip::OTAFirmwareProcessor sApplicationProcessor;
```

Comment:
why is `sApplicationProcessor` not required for `SLI_SI91X_MCU_INTERFACE`, the Si917 should also have an application processor right?

Reply:
Because 917 SoC directly support the combined image so we don't need to implement the separate processor for it

Reply:
sApplicationProcessor  is created for EFR host applications and sWifiFirmwareProcessor handles SoC and NCP OTA image upgrade.